geo-traits/src/triangle.rs

```diff
@@ -0,0 +1,111 @@
+use std::marker::PhantomData;
+
+use crate::{Dimensions, PointTrait, UnimplementedPoint};
+use geo_types::{Coord, CoordNum, Triangle};
+
+/// A trait for accessing data from a generic Triangle.
+///
+/// A triangle is a bounded area whose three vertices are defined by [points][PointTrait].
+///
+/// Refer to [geo_types::Triangle] for information about semantics and validity.
+pub trait TriangleTrait: Sized {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// The type of each underlying coordinate, which implements [PointTrait]
+    type PointType<'a>: 'a + PointTrait<T = Self::T>
+    where
+        Self: 'a;
+
+    /// The dimension of this geometry
+    fn dim(&self) -> Dimensions;
+
+    /// Access the first point in this Triangle
+    fn first(&self) -> Self::PointType<'_>;
+
+    /// Access the second point in this Triangle
+    fn second(&self) -> Self::PointType<'_>;
+
+    /// Access the third point in this Triangle
+    fn third(&self) -> Self::PointType<'_>;
+
+    /// Access the three underlying points
+    fn points(
+        &self,
+    ) -> (
+        Self::PointType<'_>,
+        Self::PointType<'_>,
+        Self::PointType<'_>,
```

Comment:
How do you feel about making this an array (`[T; 3]`) rather than a tuple so callers can call `iter()` on it to iterate over the points?